From 3f8f155df6a2e34ebdd1ab46008a4eb5c0dad9e6 Mon Sep 17 00:00:00 2001
From: manikanta kanamarlapudi <manikak@nvidia.com>
Date: Thu, 19 Jul 2012 20:50:14 +0530
Subject: [PATCH 33/40] libstagefright : extended the buffer size of avcC

This avcC atom contains the header information of decoder like sps
and pps . This header size may varies . Instead of allocating the
buffer with fixed size allocated according to the size of atom

bug 1015073

Change-Id: Ifdf3d0a63bc8c2f9a9875c78252159e4ee05958c
---
 media/libstagefright/MPEG4Extractor.cpp |   11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 871941b..8d5717a 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1337,13 +1337,15 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
 
         case FOURCC('a', 'v', 'c', 'C'):
         {
-            char buffer[512];
-            if (chunk_data_size > (off64_t)sizeof(buffer)) {
-                return ERROR_BUFFER_TOO_SMALL;
-            }
+            uint8_t *buffer = NULL;
+            buffer = (uint8_t *)malloc(chunk_data_size);
+
+            if (buffer == NULL)
+                return ERROR_MALFORMED;
 
             if (mDataSource->readAt(
                         data_offset, buffer, chunk_data_size) < chunk_data_size) {
+                free(buffer);
                 return ERROR_IO;
             }
 
@@ -1351,6 +1353,7 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
                     kKeyAVCC, kTypeAVCC, buffer, chunk_data_size);
 
             *offset += chunk_size;
+            free(buffer);
             break;
         }
 
-- 
1.7.1

